crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -1301,6 +1301,12 @@ mod private {
             &self.room_linked_chunk
         }
 
+        /// Returns a mutable reference to the underlying room linked chunk.
+        #[cfg(feature = "e2e-encryption")]
+        pub(in crate::event_cache) fn room_linked_chunk_mut(&mut self) -> &mut EventLinkedChunk {
```

Comment:
I think it breaks the purpose of `RoomEventCacheState`, not a big deal for the moment, we can still refactor later. Can we reduce the visibility scope, maybe like `pub(in super::super)?